src/api/procedures/evaluateMultiSigProposal.ts

```diff
@@ -0,0 +1,134 @@
+import { MultiSigProposal, PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, MultiSigProposalAction, ProposalStatus, TxTags } from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+import {
+  bigNumberToU64,
+  boolToBoolean,
+  signerToSignatory,
+  stringToAccountId,
+} from '~/utils/conversion';
+
+/**
+ * @hidden
+ */
+export interface MultiSigProposalVoteParams {
+  action: MultiSigProposalAction;
+  proposal: MultiSigProposal;
+}
+
+/**
+ * @hidden
+ */
+export async function prepareMultiSigProposalEvaluation(
+  this: Procedure<MultiSigProposalVoteParams, void>,
+  args: MultiSigProposalVoteParams
+): Promise<
+  | TransactionSpec<void, ExtrinsicParams<'multiSig', 'approveAsKey'>>
+  | TransactionSpec<void, ExtrinsicParams<'multiSig', 'rejectAsKey'>>
+> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: { multiSig },
+        query: {
+          multiSig: { votes },
+        },
+      },
+    },
+    context,
+  } = this;
+  const {
+    proposal: {
+      id,
+      multiSig: { address: multiSigAddress },
+    },
+    proposal,
+    action,
+  } = args;
+
+  const rawAddress = stringToAccountId(multiSigAddress, context);
```

Comment:
(nit) can this be `rawMultiSigAddress`? Makes it a bit clearer since we also have signing account to consider